lib/index.js

```diff
@@ -356,25 +360,54 @@ internals.Client.prototype.read = function (res, options, callback) {
             return callback(null, null);
         }
 
-        if (options.json === 'force') {
+        const parse = () => {
+
+            if (!options.json) {
+                return callback(err, buffer);
+            }
+
+            if (options.json === 'force') {
+                result = internals.tryParseBuffer(buffer);
+                return callback(result.err, result.json);
+            }
+
+            // mode is "smart", "strict" or true
+
+            const contentType = (res.headers && internals.findHeader('content-type', res.headers)) || '';
+            const mime = contentType.split(';')[0].trim().toLowerCase();
+
+            if (!internals.jsonRegex.test(mime)) {
+                if (options.json === 'strict') {
+                    return callback(Boom.notAcceptable('The content-type is not JSON compatible'));
+                }
+                return callback(null, buffer);
+            }
+
             result = internals.tryParseBuffer(buffer);
             return callback(result.err, result.json);
-        }
+        };
 
-        // mode is "smart", "strict" or true
+        if (options.gunzip) {
+            const contentEncoding = options.gunzip === 'force' ?
+                'gzip' :
+                (res.headers && internals.findHeader('content-encoding', res.headers)) || '';
 
-        const contentType = (res.headers && internals.findHeader('content-type', res.headers)) || '';
-        const mime = contentType.split(';')[0].trim().toLowerCase();
+            if (contentEncoding === 'gzip') {
+                Zlib.gunzip(buffer, (err, gunzipped) => {
```

Comment:
I would much prefer to use streaming gunzip, rather than the sync one that you use. Eg. create a passthrough gunzip stream and pipe through it. This might even be a simpler implementation.